### PR TITLE
Fix prometheus-postprocess script to write all the collected metrics in to the file

### PR DESCRIPTION
**Pull request by chaitanyaenr**
- Prometheus-metrics-postprocess script is not writing all the
  metrics to the output file even though it does the postprocessing
  on the collected data. This commit fixes that bug.
- This commit also adds checks to verify inventory variable is
  defined, if yes it checks if the file exists and is not empty.

### CODE REVIEW

**Comment by chaitanyaenr** [comment]
@ndokos PTAL.

File: agent/tool-scripts/prometheus-metrics [hunk 1]
```diff
@@ -115,6 +114,11 @@ tool_stderr_file=$tool_output_dir/$tool-stderr.txt
 tool_log=$tool_output_dir/tool.log
 ## create directory to save the keys from openshift master, hosts
 mkdir -p /run/pbench/keys
+## check if the inventory variable is set
+if [[ -z "$inventory" ]] || [[ -s "$inventory" ]]; then
```
Comment (ndokos) [inline agent/tool-scripts/prometheus-metrics]: The check is still wrong I thinK: -s checks for size greater than 0, in which case you don't want to complain. You should combine all the conditions into a single test too: `[[ -z "$inventory" || ! -s "$inventory" ]]`. But don't trust me: check that it is right! Or you might want to break the check up into if/elif/... cases and make each complaint more specific.

**Comment by chaitanyaenr** [comment]
@ndokos PTAL.

File: agent/tool-scripts/prometheus-metrics [hunk 2]
```diff
@@ -115,6 +114,11 @@ tool_stderr_file=$tool_output_dir/$tool-stderr.txt
 tool_log=$tool_output_dir/tool.log
 ## create directory to save the keys from openshift master, hosts
 mkdir -p /run/pbench/keys
+## check if the inventory variable is set
+if [[ -z "$inventory" ]] || [[ ! -s "$inventory" ]]; then
```
Comment (ndokos) [inline agent/tool-scripts/prometheus-metrics]: What if they have set $inventory to a non-empty *directory*?

File: agent/tool-scripts/prometheus-metrics [hunk 3]
```diff
@@ -115,6 +114,11 @@ tool_stderr_file=$tool_output_dir/$tool-stderr.txt
...
+	echo "Please set the inventory=/path/to/inventory while registering the tool and make sure file exists and is not empty"
```
Comment (ndokos) [inline agent/tool-scripts/prometheus-metrics]: It would be helpful I think if you mentioned *what* file is missing or empty in the error message.

**Comment by chaitanyaenr** [comment]
@ndokos PTAL.

**Comment by chaitanyaenr** [comment]
Fixes #597

**Pull request merged**